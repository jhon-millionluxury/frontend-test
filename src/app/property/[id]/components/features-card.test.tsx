/* eslint-disable @typescript-eslint/no-explicit-any */
import { render, screen } from "@testing-library/react";
import PropertyDetailPageFeaturesCard from "@/app/property/[id]/components/features-card";

// Mockear dependencias UI para simplificar
jest.mock("../../../../components/ui/card", () => ({
  Card: ({ children, ...props }: any) => (
    <div data-testid="card" {...props}>
      {children}
    </div>
  ),
  CardContent: ({ children, ...props }: any) => (
    <div data-testid="card-content" {...props}>
      {children}
    </div>
  ),
}));

jest.mock("../../../../components/ui/badge", () => ({
  Badge: ({ children, ...props }: any) => (
    <span data-testid="badge" {...props}>
      {children}
    </span>
  ),
}));

describe("PropertyDetailPageFeaturesCard", () => {
  it("renders heading correctly", () => {
    render(<PropertyDetailPageFeaturesCard features={["Pool", "Garage"]} />);
    expect(screen.getByText("Premium Features")).toBeInTheDocument();
  });

  it("renders a badge for each feature", () => {
    const features = ["Pool", "Garage", "Garden"];
    render(<PropertyDetailPageFeaturesCard features={features} />);

    features.forEach((feature) => {
      expect(screen.getByText(feature)).toBeInTheDocument();
    });

    expect(screen.getAllByTestId("badge")).toHaveLength(features.length);
  });

  it("handles empty features gracefully", () => {
    render(<PropertyDetailPageFeaturesCard features={[]} />);

    expect(screen.getByText("Premium Features")).toBeInTheDocument();

    expect(screen.queryByTestId("badge")).not.toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <PropertyDetailPageFeaturesCard features={["Balcony", "WiFi"]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
