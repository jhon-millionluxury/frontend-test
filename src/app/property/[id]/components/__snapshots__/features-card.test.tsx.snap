// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PropertyDetailPageFeaturesCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="card"
  >
    <div
      class="p-6"
      data-testid="card-content"
    >
      <h2
        class="mb-4 font-serif text-2xl font-semibold text-foreground"
      >
        Premium Features
      </h2>
      <div
        class="flex flex-wrap gap-2"
      >
        <span
          class="px-4 py-2 text-sm"
          data-testid="badge"
          variant="secondary"
        >
          Balcony
        </span>
        <span
          class="px-4 py-2 text-sm"
          data-testid="badge"
          variant="secondary"
        >
          WiFi
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
